src/extension/src/azure/utils/nameGenerator.ts

```diff
@@ -40,12 +40,22 @@ export namespace NameGenerator {
 
   export function generateName(
     userProjectName: string,
-    resourceType?: string
+    azureType?: AzureResourceType
   ): string {
+    let result;
     const timestamp = unixToSuffix(Date.now());
-    const suffix =
-      resourceType === undefined ? timestamp : resourceType + "-" + timestamp;
-    return userProjectName + "-" + suffix;
+    result = userProjectName + "-" + timestamp;
+    if (
```

Comment:
Wondering if you should preserve the timestamp at all times since that is the uniqueness factor. Currently you strip away the timestamp if the name is too long! Think of the following situation: You select a CosmosDb name with length 31. You will be retrying the same name over and over again. That is not ideal. Consider another case where the length is around 27 you would be stripping away most of the date, hours minutes and seconds. 

Another idea would be to put a max length on the app name length as well!

Reply:
I also think we should preserve the timestamp and trimming the project name

Reply:
@SahilTara thanks for pointing it out, it totally make sense